fix(pandas-postprocessing): resolve SettingWithCopyWarning in histogram and boxplot

nitishagar opened the pull request:
## **User description**
### SUMMARY

  Fix the Pandas `SettingWithCopyWarning` that appears in logs when loading dashboards with Histogram and Box Plot charts.

  **Root Cause:** When `df.dropna()` is called, Pandas cannot guarantee whether it returns a view or a copy. Subsequent direct column assignment `df[column] = value` on a potential view triggers the warning.

  **Fix:** Use the `.loc[]` accessor for column assignment (`df.loc[:, column] = value`), which is the Pandas-recommended approach to avoid this warning.

  **Changes:**
  - `superset/utils/pandas_postprocessing/histogram.py:57` - Use `.loc[]` accessor
  - `superset/utils/pandas_postprocessing/boxplot.py:130` - Use `.loc[]` accessor (same pattern)

  This is a cosmetic fix that eliminates warnings without changing any functionality.

  ### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

  **Before (in logs):**
  superset_app | /app/superset/utils/pandas_postprocessing/histogram.py:57: SettingWithCopyWarning:
  superset_app | A value is trying to be set on a copy of a slice from a DataFrame.
  superset_app | Try using .loc[row_indexer,col_indexer] = value instead

  **After:** No warning in logs.

  ### TESTING INSTRUCTIONS

  1. Start Superset: `docker compose up`
  2. Login at `http://localhost:8088` with admin credentials
  3. Create or load a dashboard with a Histogram chart
  4. Check logs: `docker compose logs superset_app | grep SettingWithCopyWarning`
  5. Confirm no warning appears

  **Automated tests:**
  ```bash
  pytest tests/unit_tests/pandas_postprocessing/test_histogram.py tests/unit_tests/pandas_postprocessing/test_boxplot.py -v
  All 16 tests pass.
 ```

  **ADDITIONAL INFORMATION**

  - Has associated issue: Fixes #36530
  - Required feature flags:
  - Changes UI
  - Includes DB Migration (follow approval process in https://github.com/apache/superset/issues/13351)
    - Migration is atomic, supports rollback & is backwards-compatible
    - Confirm DB migration upgrade and downgrade tested
    - Runtime estimates and downtime expectations provided
  - Introduces new feature or API
  - Removes existing feature or API


___

## **CodeAnt-AI Description**
**Stop pandas warnings when preparing histogram and box plot data**

### What Changed
- Histogram values are now converted without triggering a pandas warning after empty rows are removed
- Box plot data now converts object-type columns safely before calculating summary values
- These chart outputs keep the same results, but log noise from pandas warnings is removed

### Impact
`✅ Cleaner dashboard logs`
`✅ Fewer pandas warning alerts`
`✅ More reliable chart loading`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

rusackas commented:
Let us know if you want to address the bot suggestions. Thanks for the contribution!

nitishagar committed:
fix(pandas-postprocessing): resolve SettingWithCopyWarning in histogram and boxplot

Use df.loc[:, column] syntax and is_object_dtype() to avoid pandas
SettingWithCopyWarning when modifying DataFrame columns in place.

rusackas committed:
Merge branch 'master' into fix-36530-histogram-warning
